pyrealm/constants/core_const.py

```diff
@@ -0,0 +1,251 @@
+    kg/mol)"""
+    k_CtoK: float = 273.15
+    """Conversion from °C to K   (:math:`CtoK` , 273.15, -)"""
+    k_pir = np.pi / 180.0
+    """Conversion factor from radians to degrees   (``pir`` , ~0.01745, -)"""
+
+    # TODO - these might be better in a separate SplashConst
+    k_w = 0.26
+    """Entrainment factor (Lhomme, 1997; Priestley & Taylor, 1972)"""
+    k_Cw = 1.05
+    """Supply constant, mm/hr (Federer, 1982)"""
+
+    # Solar constants
+    k_A = 107.0  # constant for Rnl (Monteith & Unsworth, 1990)
+    k_alb_sw = 0.17  # shortwave albedo (Federer, 1968)
+    k_alb_vis = 0.03  # visible light albedo (Sellers, 1985)
+    k_b = 0.20  # constant for Rnl (Linacre, 1968)
+    k_c = 0.25  # cloudy transmittivity (Linacre, 1968)
+    k_d = 0.50  # angular coefficient of transmittivity (Linacre, 1968)
+    k_fFEC = 2.04  # from flux to energy conversion, umol/J (Meek et al., 1984)
+    k_Gsc = 1360.8  # solar constant, W/m^2 (Kopp & Lean, 2011)
+
+    # Paleoclimate variables:
+    ke = 0.0167  # eccentricity for 2000 CE (Berger, 1978)
+    keps = 23.44  # obliquity for 2000 CE, degrees (Berger, 1978)
+    komega = 283.0  # longitude of perihelion for 2000 CE, degrees (Berger, 1978)
+
+    # Paleoclimate variables:
+    k_e: float = 0.0167
+    """Solar eccentricity, using default value for 2000 CE :cite:t:`berger:1978a`."""
+    k_eps: float = 23.44
+    """Solar obliquity, using default value for 2000 CE :cite:t:`berger:1978a`."""
+    k_omega = 283.0
+    """Solar longitude of perihelion, using default value for 2000 CE
+    :cite:t:`berger:1978a`."""
+
+    # Hygro constants
+    magnus_coef: NDArray[np.float32] = field(
+        default_factory=lambda: np.array((611.2, 17.62, 243.12))
+    )
+    """Three coefficients of the Magnus equation for saturated vapour pressure,
+    defaulting to those of  ``Sonntag1990``."""
+    mwr: float = 0.622
+    """The ratio molecular weight of water vapour to dry air (:math:`MW_r`, -)"""
+    magnus_option: Optional[str] = None
+    """Pre-defined Magnus equation parameterisations. Use one of ``Allen1998``,
+    ``Alduchov1996`` or ``Sonntag1990`` to set values for
+    :attr:`~pyrealm.constants.core_const.CoreConst.magnus_coef`.
+    """
+
+    # Water constants
+    water_density_method: str = "fisher"
+    """Set the method used for calculating water density ('fisher' or 'chen')."""
+
+    # Fisher Dial
+    fisher_dial_lambda: NDArray[np.float32] = field(
+        default_factory=lambda: np.array(
+            [1788.316, 21.55053, -0.4695911, 0.003096363, -7.341182e-06]
+        )
+    )
+    r"""Temperature dependent lambda parameterisation of the Tumlirz equation.
+     (:math:`\lambda`)."""
+
+    fisher_dial_Po: NDArray[np.float32] = field(
+        default_factory=lambda: np.array(
+            [5918.499, 58.05267, -1.1253317, 0.0066123869, -1.4661625e-05]
+        )
+    )
+    """Temperature dependent P0 parameterisation of the Tumlirz equation
```

Comment:
What is P0? Might be worth writing out here because this might be read out of context.